### DISCUSSION TITLE
Difficulty configuring/installing R on mox

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1178] [Q&A] [answered]
I am following [these instructions](https://wiki.cac.washington.edu/display/hyakusers/Hyak+R+programming) to get the latest version of R on mox using source code (The UC Berkeley CRAN mirror link is defunct, so I am using one from [OSU](https://ftp.osuosl.org/pub/cran/)).

I'm on step 9, configuration/installation:

<img width="614" alt="Screen Shot 2021-04-13 at 1 04 26 PM" src="https://user-images.githubusercontent.com/22335838/114613945-c9b3f000-9c58-11eb-9119-e2ed9d1126eb.png">

When I try running the configure command, I get the following error:

<img width="748" alt="Screen Shot 2021-04-13 at 1 04 53 PM" src="https://user-images.githubusercontent.com/22335838/114613999-d8020c00-9c58-11eb-99cd-64ee9ff71469.png">

Is there something I'm missing?

**Comment by sr320** [comment]
feels like you are in wrong directory... see step 7?

**Comment by kubu4** [comment]
If you're following those instructions exactly, you need to be in the `Rstuff` directory you created in Step 2.

Basically, I suspect you're in the wrong directory. You need to be in the same directory in which you downloaded the `R*.gz` file.

**Reply by kubu4** [reply]
And then, like @sr320 points out, you need to change in to the directory created in Step 7.

**Reply by yaaminiv** [reply]
yup, I was in the wrong directory! 

I forgot that I was unable to download into the `Rstuff` directory so I downloaded it into my user directory, then navigated out back into the wrong directory when I did step 8

**Comment by yaaminiv** [comment]
Okay, new question!

I ran `./configure --prefix=/gscratch/srlab/yaaminiv/Rinstall` from the `R-4.0.5` directory. The `R-4.0.5` directory is NOT in `Rstuff`, but in my user directory.  I got the following error:

<img width="748" alt="Screen Shot 2021-04-13 at 1 18 32 PM" src="https://user-images.githubusercontent.com/22335838/114615606-c02b8780-9c5a-11eb-82a4-aee4e7ab4633.png">

`configure: error: PCRE2 library and headers are required, or use --with-pcre1 and PCRE >= 8.32 with UTF-8 support`

A quick search lead me to [this page](https://unix.stackexchange.com/questions/635758/local-installation-of-pcre2-not-detected-while-installing-r-4-0-4-from-source), so I think I need to install this package? However, I was looking through the R installation instructions again, and I saw that having anaconda python would prevent successful installation.

<img width="1070" alt="Screen Shot 2021-04-13 at 1 22 51 PM" src="https://user-images.githubusercontent.com/22335838/114616122-5b246180-9c5b-11eb-8efe-d0617a97571e.png">

How do I remove anaconda python from my path? I'm looking at my ~/.bashrc profile...do I just delete `export PATH="/usr/lusers/yaaminiv/miniconda3/bin:$PATH`?

<img width="749" alt="Screen Shot 2021-04-13 at 1 27 46 PM" src="https://user-images.githubusercontent.com/22335838/114616716-0a613880-9c5c-11eb-94f3-ed7847dd7528.png">


*edited to include full error message*

**Reply by kubu4** [reply]
First, I'd check if miniconda is in your $PATH. Along those lines, I noticed in your original post, you were already in an Anaconda environment (denoted by the `(base)` in front of your shell info).

So, do this first, in a shell (exit Nano, if you're still in that):

1. `conda deactivate`

2. `which conda`

Please report results of `which conda`.

**Reply by yaaminiv** [reply]
<img width="749" alt="Screen Shot 2021-04-13 at 1 53 47 PM" src="https://user-images.githubusercontent.com/22335838/114619684-acceeb00-9c5f-11eb-98e3-b806502a41a3.png">

**Reply by kubu4** [reply]
Alternatively, did you try the suggestion in the error message (or use --pcre1)? The message is cutoff, so we cant' see the full suggestion.

You can also try adding `--enable-jit` to your `./configure` command.

Also, you may need to load a GCC compiler module to get this to install properly. See the "A.1 Essential programs and libraries" section of compiling R for more info on the things I mentioned:

https://cran.r-project.org/doc/manuals/r-devel/R-admin.html

**Reply by kubu4** [reply]
Okay, if you want to follow the suggestions in those instructions you posted above, I'd comment out all of that `conda initialize` section. Don't delete anything, though. After commenting stuff out, follow the rest of the instructions regarding logging out, etc.

**Reply by yaaminiv** [reply]
Whoops, just edited to include the full error message.

Tried `./configure --prefix=/gscratch/srlab/yaaminiv/Rinstall --with-prce1` and got the same error message.

**Reply by yaaminiv** [reply]
- Commented out `conda initialize` section
- Logged out and requested a new build node
- Ran `./configure --prefix=/gscratch/srlab/yaaminiv/Rinstall`, same error
- Ran `./configure --prefix=/gscratch/srlab/yaaminiv/Rinstall --with-prce1`successfully (I think!)! Got some warnings:

<img width="749" alt="Screen Shot 2021-04-13 at 2 09 17 PM" src="https://user-images.githubusercontent.com/22335838/114621678-d852d500-9c61-11eb-8661-5aba95036889.png">

- Ran `make` with no error
- Ran `make install`, ended up with disk quota error.  
- 
<img width="1120" alt="Screen Shot 2021-04-13 at 2 27 12 PM" src="https://user-images.githubusercontent.com/22335838/114623620-587a3a00-9c64-11eb-8adb-1c0d3c576544.png">

How do I get more disk space? I only have R installation stuff + miniconda in my user directory.

**Reply by kubu4** [reply]
You can't get more space on your login node. That's why we have the programs directory in the `/gscratch/srlab/programs/`.

However, we're also over quota there... Have deleted some stuff, but takes time to update the system.

**Reply by yaaminiv** [reply]
> You can't get more space on your login node. That's why we have the programs directory in the /gscratch/srlab/programs/.

It's installing at `/gscratch/srlab/yaaminiv/Rinstall/`, not my login node! So at least that's good.

>However, we're also over quota there... Have deleted some stuff, but takes time to update the system.

I'll wait and see if I can try installing again.

**Comment by sr320** [comment]
I might be missing something but cannot you just load the module...
partial list of what is available
```
r_3.3.3                                                      
r_3.5.1                                                      
r_3.6.0                                                      
r_3.6.0+Rmpi-impi_2019   
```

you are installing from source (which might not be necessary?)

**Reply by kubu4** [reply]
I think she wanted a newer version, as we also have `/gscratch/srlab/programs/R-3.6.2/`

**Reply by sr320** [reply]
OK 
FWIW this works
```
[sr320@n2193 R]$ module load r_3.6.0
[sr320@n2193 R]$ module list
Currently Loaded Modulefiles:
 1) icc_19   2) r_3.6.0  
[sr320@n2193 R]$ R

R version 3.6.0 (2019-04-26) -- "Planting of a Tree"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> 
```

**Reply by yaaminiv** [reply]
@sr320  I wanted the newest version of R (if possible)! I'll see if dealing with conda lets me load R-4.0.5, and if not, I'll test the workflow with the versions of R we already have installed

**Comment by yaaminiv** [comment, marked as answer]
<img width="582" alt="Screen Shot 2021-04-13 at 3 38 00 PM" src="https://user-images.githubusercontent.com/22335838/114629894-3f768680-9c6e-11eb-897e-bee03a891e55.png">

Success!! 

Tips in case anyone has to do this again......

- use python, not anaconda python
- download R to user directory and configure/make/install in `/gscratch/srlab/YOURDIRECTORY/Rinstall`
- make sure there is enough disk space
